### PR TITLE
Pybind11

### PR DESCRIPTION
**Pull request by JensGM**

### CODE REVIEW

**Comment by ReedOnly** [comment]
retest this please

**Comment by JensGM** [comment]
retest this please

File: lib/src/nexus_plot.cpp [hunk 1]
```diff
@@ -34,6 +34,7 @@
 #include <winsock2.h>
 #endif
 
+#include <ert/ecl/ecl_sum.h>
```
Comment (pgdr) [inline lib/src/nexus_plot.cpp:37]: why added?

Comment (JensGM) [reply lib/src/nexus_plot.cpp:37]: That header is no longer included in nexus_plot.hpp, so it is needed here now.

Comment (pgdr) [reply lib/src/nexus_plot.cpp:37]:
```
nex/python/nex/nex.cpp:1:31: fatal error: pybind11/pybind11.h: No such file or directory
 #include <pybind11/pybind11.h>
                               ^
compilation terminated.
error: command 'gcc' failed with exit status 1
python/CMakeFiles/nex-python.dir/build.make:57: recipe for target 'python/CMakeFiles/nex-python' failed
make[2]: *** [python/CMakeFiles/nex-python] Error 1
CMakeFiles/Makefile2:1247: recipe for target 'python/CMakeFiles/nex-python.dir/all' failed
make[1]: *** [python/CMakeFiles/nex-python.dir/all] Error 2
Makefile:138: recipe for target 'all' failed
make: *** [all] Error 2
```

Comment (pgdr) [reply lib/src/nexus_plot.cpp:37]: Should `<pybind11/pybind11.h>` be present?   If so, why do Travis pass?

Comment (JensGM) [reply lib/src/nexus_plot.cpp:37]: You need to install pybind11 with pip, or install requirements with pip

Comment (pgdr) [reply lib/src/nexus_plot.cpp:37]: Which argument to `cmake` should tell `cmake` where to find the header file?

**Comment by pgdr** [comment]
```patch
 CMakeLists.txt     | 2 ++
 python/setup.py.in | 1 +
 2 files changed, 3 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index e432b3e..a7adfd0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,6 +1,8 @@
 cmake_minimum_required(VERSION 2.8.12)
 project(nex)
 
+set(PYBIND11_INCLUDE_DIR "" CACHE PATH "Location to search for pybind11 headers")
+
 include(CheckFunctionExists)
 include(CheckIncludeFile)
 include(CTest)
diff --git a/python/setup.py.in b/python/setup.py.in
index d0888e5..e804a01 100644
--- a/python/setup.py.in
+++ b/python/setup.py.in
@@ -16,6 +16,7 @@ pynex = Extension(
     ['${CMAKE_CURRENT_SOURCE_DIR}/nex/nex.cpp'],
     include_dirs=[
         '${NEX_INCLUDE_DIR}',
+        '${PYBIND11_INCLUDE_DIR}',
         get_pybind_include(),
         get_pybind_include(user=True)
     ],
```

**Comment by pgdr** [comment]
```patch
 CMakeLists.txt     |  2 ++
 python/setup.py.in | 13 +++++++++++--
 2 files changed, 13 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index e432b3e..a7adfd0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,6 +1,8 @@
 cmake_minimum_required(VERSION 2.8.12)
 project(nex)
 
+set(PYBIND11_INCLUDE_DIR "" CACHE PATH "Location to search for pybind11 headers")
+
 include(CheckFunctionExists)
 include(CheckIncludeFile)
 include(CTest)
diff --git a/python/setup.py.in b/python/setup.py.in
index d0888e5..b643122 100644
--- a/python/setup.py.in
+++ b/python/setup.py.in
@@ -4,11 +4,18 @@ import sys
 import setuptools
 
 class get_pybind_include(object):
-    def __init__(self, user=False):
+    def __init__(self, user=False, relative=False):
         self.user = user
+        self.rel = relative
 
     def __str__(self):
         import pybind11
+        if self.rel:
+            import os.path as op
+            prefix = op.dirname(pybind11.__file__)
+            postfix = op.join('..', '..', '..', '..', 'include', 'python2.7')
+            return op.realpath(op.join(prefix, postfix))
+
         return pybind11.get_include(self.user)
 
 pynex = Extension(
@@ -16,8 +23,10 @@ pynex = Extension(
     ['${CMAKE_CURRENT_SOURCE_DIR}/nex/nex.cpp'],
     include_dirs=[
         '${NEX_INCLUDE_DIR}',
+        '${PYBIND11_INCLUDE_DIR}',
         get_pybind_include(),
-        get_pybind_include(user=True)
+        get_pybind_include(user=True),
+        get_pybind_include(relative=True),
     ],
     libraries=['nex'],
     library_dirs=['${NEX_LIBRARY_DIR}'],
```

**Review by pgdr — APPROVED** [review]
Good work, LGTM!

Squash and merge at will!

**Pull request merged**